code/modules/vending/vending.dm

```diff
@@ -1958,7 +1958,7 @@
 				R.product_amount += 1
 				existed = TRUE
 				break
-			else if (istype(target,R.product_type) && R.real_name == target.real_name)
+			else if (istype(target,R.product_type) && (R.real_name || R.product_name) == (target.real_name || target.name))
```

Comment:
I don't believe `||` expressions are converted to booleans in DM. According to the [reference](https://secure.byond.com/docs/ref/#/operator/||) and testing, `||` appears to just return the first true value in the expression if there is one. So like:
```dm
("foo" || null) // returns "foo"
(null || "bar") // returns "bar"
("foo" || "bar") // returns "foo"
(null || null) // returns null
```
So it just means that if `R.real_name == target.name` or `R.product_name == target.real_name`, it would also work.
If you don't like that behavior though, I'll just use your suggestion.

Reply:
Huh, funky. Thanks for the knowledge.

Reply:
> Huh, funky. Thanks for the knowledge.

Called short-circuiting